### PR TITLE
update Capabilities

### PR DESCRIPTION
**Pull request by jasperRob**

### CODE REVIEW

File: qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java [hunk 1]
```diff
@@ -165,39 +186,23 @@ public boolean hasCapability(final String rawCapabilityCode, final CapabilityMod
...
-					String roleCode = roleArray.getString(i);
-
-					CapabilityMode[] modes = getCapabilitiesFromCache(roleCode, cleanCapabilityCode);
-					List<CapabilityMode> modeList = Arrays.asList(modes);
-					for (CapabilityMode checkMode : checkModes) {
-						if (!modeList.contains(checkMode))
-							return false;
-					}
+				CapabilityMode[] modes = getCapabilitiesFromCache(code, cleanCapabilityCode);
+				List<CapabilityMode> modeList = Arrays.asList(modes);
```
Comment (varunshastry) [inline qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java]: Change from List to HashSet to get better performance.

File: qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java [hunk 2]
```diff
@@ -117,20 +135,23 @@ public BaseEntity addCapabilityToBaseEntity(BaseEntity targetBe, final String ra
...
 
-		JsonObject object = jsonb.fromJson(cachedObject, JsonObject.class);
+		String productCode = userToken.getProductCode();
+		String key = String.format("%s:%s", targetCode, capabilityCode);
 
-		if ("error".equals(object.getString("status"))) {
-			log.error("Error reading cache for realm: " + productCode + " with key: " + key);
-			return null;
-		}
+		CapabilityMode[] modes = CacheUtils.getObject(productCode, key, CapabilityMode[].class);
```
Comment (varunshastry) [inline qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java]: We could change Array to HashSet here.

**Review by Expressionless — APPROVED** [review]
I don't see anything wrong with this, and I assume Jasper has gone through and tested the code. LGTM

**Review by varunshastry — CHANGES_REQUESTED** [review]

File: qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java (hunk 1 above)
Comment (jasperRob) [reply qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java]: Changed to HashSet

File: qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java (hunk 2 above)
Comment (jasperRob) [reply qwandaq/src/main/java/life/genny/qwandaq/utils/CapabilityUtils.java]: Changed to HashSet

**Review by varunshastry — APPROVED** [review]

**Pull request merged**